src/Types.jl

```diff
@@ -892,52 +893,251 @@ function ensure_resolved(env::EnvCache,
     pkgerror(msg)
 end
 
-const DEFAULT_REGISTRIES = Dict("General" => "https://github.com/JuliaRegistries/General.git")
+##############
+# Registries #
+##############
+
+mutable struct RegistrySpec
+    name::Union{String,Nothing}
+    uuid::Union{UUID,Nothing}
+    url::Union{String,Nothing}
+    # the path field can be a local source when adding a registry
+    # otherwise it is the path where the registry is installed
+    path::Union{String,Nothing}
```

Comment:
I'm confused about the purpose of `path` here: is it being used for two different things?

Reply:
Yea, it can play the role of the path to where we add it from (if adding from a path) but most often this is just the path where we read the registry from. I am not sure we should allow adding registries from paths, if they are not git repos though.